presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanMatchPattern.java

```diff
@@ -222,19 +209,11 @@ public static SymbolStem symbolStem(String stem)
         return new SymbolStem(stem);
     }
 
-    public static class SymbolStem extends Symbol
+    private interface Stem
     {
-        private final String stem;
-
-        SymbolStem(String stem)
-        {
-            // For useful output when we print out failure messages.
-            super(stem + "_*");
-            this.stem = stem;
-        }
+        String getStem();
 
-        @Override
-        public boolean equals(Object o)
+        default <T> boolean stemEquals(Object o, Class<T> clazz, Function<T, String> getName)
```

Comment:
Why `Function<T, String> getName`?
Isn't it easier and less error-prone to declare e.g. `abstract String getName()` which will be implemented inherited in `SymbolSteam` from `Symbol`? Then you don't even need overriding `equals` in sub-classes.

Or if you follow my suggestion about _composition over inheritance_ you can provide a `Symbol delegate()` abstract method here which will be implemented by `SymbolStem` and `SymbolReferenceStem`.